### ISSUE TITLE
Adjustable Text Spacing (low priority)

### ISSUE THREAD

**Issue by manfromjupyter** [jupyterlab/jupyterlab#9684]
## Purpose:
Given WCAG requirement 1.4.12, need to add adjustable text spacing (for people with low vision handicaps that use various tools or browsers extensions to help them use the software)

## Notes
This is low priority as it is not needed for WCAG 2.0 compliance, but is needed for WCAG 2.1 compliance. Only CSS knowledge should be needed to fix and is not dependent on the other accessibility tickets so can be done any time. In addition ticket does not depend on others so can be done synchronously with the others.  

Original source of ticket: #9399

## Adjustable text spacing
> Expectation: The line height, paragraph spacing, letter spacing, and word spacing can be adjusted without loss of content or functionality. Should support line height at least 1.5 times font-size, letter spacing 0.12 times font-size, paragraph spacing at least 2 times font-size, and word-spacing at least 0.16 times font-size. (WCAG Criteria 1.4.12 Text Spacing).
Observed: When injecting the CSS most accessibility or elder-friendly browser extensions inject, found that Launcher tab’s text within kernel containers wraps and gets cut off (accessibility issue). In addition the nav in the header gets off centered and leaves a shadow looking thing when expanding their dropdowns (cosmetic issue).

**Comment by saulshanabrook** [comment]
Thanks, @manfromjupyter for reporting this! I was curious if you had any tips on trying to reproduce your screenshot to test fixes.

**Comment by manfromjupyter** [comment]
@saulshanabrook shoot I'm so sorry, I did not see this when you commented it, I'm only here by coincidence. I should have included exactly how I reproduced it and that's actually why I came here... to make sure I did.

Add the following CSS to the site.
```
* {
    line-height: 1.5 !important;
    letter-spacing: 0.12em !important;
    word-spacing: 0.16em !important;
}
```
## 2 Ways to do this
To do this open up your Inspect Element tool by right clicking on the page and then selecting "Inspect" or "Inspect Element" (recommend the Chrome browser). After that there are 2 ways to do this.

### Way # 1
On any element click on it and over on the right side click on the `+` (plus) icon to add new style. Once that populates, change the text before the `{ }` icons to just be `*`. In CSS this selector means "any" so selecting everything. Once you apply that, in the center part, copy/paste the following:
```
    line-height: 1.5 !important;
    letter-spacing: 0.12em !important;
    word-spacing: 0.16em !important;
```
![adding new selector](https://user-images.githubusercontent.com/51969207/110863435-adeeb000-828e-11eb-8b98-0688c5521430.PNG)

Upon clicking away or pressing enter, the changes should settle in instantly.

### Way # 2
The control freak way (precise developer), go to the HTML <head></head> element at the very top where all of the other stylesheets are, right click on it and select "Edit in HTML". Once open, at the very top after the "<head>" text or bottom before the "</head>", copy and paste the following then close it up.
```
<style>* {
    line-height: 1.5 !important;
    letter-spacing: 0.12em !important;
    word-spacing: 0.16em !important;
}</style>
```
![adding style tag to html head](https://user-images.githubusercontent.com/51969207/110863427-a9c29280-828e-11eb-8b16-5fb951328b28.PNG)

**Comment by manfromjupyter** [comment]
I would subscribe a tool but I don't know of any I'd recommend and this is relatively easy if you have the code off hand or are a CSS guru. Believe the reasoning for this WCAG 2.1 requirement was to encompass a number of tools users may or already use with just mild low vision handicaps.
